torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,359 @@
+
+    Returns
+        A list of the subset datasets.
+
+    .. versionadded:: 0.5
+    """
+    if sum(fractions) != 1:
+        raise ValueError("Sum of input fractions must equal 1.")
+
+    if any(n <= 0 for n in fractions):
+        raise ValueError("All items in input fractions must be greater than 0.")
+
+    if grid_size < 2:
+        raise ValueError("Input grid_size must be greater than 1.")
+
+    new_indexes = [
+        Index(interleaved=False, properties=Property(dimension=3)) for _ in fractions
+    ]
+
+    lengths = _fractions_to_lengths(fractions, len(dataset) * grid_size**2)
+
+    cells = []
+
+    for i, hit in enumerate(
+        dataset.index.intersection(dataset.index.bounds, objects=True)
+    ):
+        minx, maxx, miny, maxy, mint, maxt = hit.bounds
+
+        stridex = (maxx - minx) / grid_size
+        stridey = (maxy - miny) / grid_size
+
+        cells.extend(
+            [
+                (
+                    (
+                        minx + x * stridex,
+                        minx + (x + 1) * stridex,
+                        miny + y * stridey,
+                        miny + (y + 1) * stridey,
+                        mint,
+                        maxt,
+                    ),
+                    hit.object,
+                )
+                for x in range(grid_size)
+                for y in range(grid_size)
+            ]
+        )
+
+    cells = [cells[i] for i in randperm(len(cells), generator=generator)]
+
+    for i, length in enumerate(lengths):
+        for j in range(length):
+            cell = cells.pop()
+            new_indexes[i].insert(j, cell[0], cell[1])
+
+    new_datasets = []
+    for index in new_indexes:
+        ds = deepcopy(dataset)
+        ds.index = index
+        new_datasets.append(ds)
+
+    return new_datasets
+
+
+def roi_split(dataset: GeoDataset, rois: Sequence[BoundingBox]) -> List[GeoDataset]:
+    """Split a GeoDataset intersecting it with a ROI for each desired new GeoDataset.
+
+    Args:
+        dataset: dataset to be split
+        rois: regions of interest of splits to be produced
+
+    Returns
+        A list of the subset datasets.
+
+    .. versionadded:: 0.5
+    """
+    new_indexes = [
+        Index(interleaved=False, properties=Property(dimension=3)) for _ in rois
+    ]
+
+    for i, roi in enumerate(rois):
+        if any(roi.intersects(x) and (roi & x).area > 0 for x in rois[i + 1 :]):
+            raise ValueError("ROIs in input rois can't overlap.")
+
+        j = 0
+        for hit in dataset.index.intersection(tuple(roi), objects=True):
+            box = BoundingBox(*hit.bounds)
+            new_box = box & roi
+            if new_box.area > 0:
+                new_indexes[i].insert(j, tuple(new_box), hit.object)
+                j += 1
+
+    new_datasets = []
+    for index in new_indexes:
+        ds = deepcopy(dataset)
+        ds.index = index
+        new_datasets.append(ds)
+
+    return new_datasets
+
+
+def time_series_split(
```

Comment:
@nilsleh can you review this one? I know you do a lot of time-series analysis work. Basically wondering if this seems like a good all-purpose technique for time-series splitting, or if there are so many different ways someone might want to perform a time-series split that we need more specialized functions.

One that thing that's never been clear to me (in general, not talking about this PR) is how to integrate time-series analysis (where you might want to do rolling window predictions) into something like PyTorch Lightning (where everything is designed for train-val-test splits).